app/services/stage_update_service.rb

```diff
@@ -62,10 +55,24 @@ def request_finished(last_decision, last_reason)
   end
 
   def stage_finished(decision)
+    if external_signal?
+      signal_external_approval_process(decision)
+    else
+      auto_next_stage(decision)
+    end
+  end
+
+  def auto_next_stage(decision)
     next_stage = stage.request.stages.find { |s| s.state == Stage::PENDING_STATE }
     return unless next_stage
 
     operation = decision == Stage::DENIED_STATUS ? Action::SKIP_OPERATION : Action::NOTIFY_OPERATION
     ActionCreateService.new(next_stage.id).create(:operation => operation, :processed_by => 'system')
   end
+
+  def signal_external_approval_process(decision)
+    template = stage.request.workflow.template
+    processor_class = "#{template.signal_setting['processor_type']}_process_service".classify.constantize
```

Comment:
Same here.  Slightly different lookup, but would be nice as a separate method at some point.